### PR TITLE
INTLY-6505 3scale 2.8

### PR DESCRIPTION
**Pull request by KevFan**
# Description

Upgrade 3scale version to version 2.8

Jira:
* https://issues.redhat.com/browse/INTLY-6505
* https://issues.redhat.com/browse/INTLY-6506

There is a known issue that is preventing the completion of walkthrough 4 on 3scale 2.8:
* https://issues.redhat.com/browse/THREESCALE-4851

### Install
* Install RHMI from this branch
* Verify 3scale sucessfully installs

### Upgrade
* Checkout master branch 
* Install RHMI 
* Verify 3scale installs successfully 
* Stop running the operator locally
* Checkout this branch
* Run the operator
* Observe that there will be an upgrade for 3scale and that it will install and complete successfully

### Install via OLM
* Change makefile `ORG` to your own quay.io registry
  * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/Makefile#L3
* Build and push RHMI image to your own registry
  * `make image/build/push`
* Replace image in csv with built image
  * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/deploy/olm-catalog/integreatly-operator/integreatly-operator-2.0.0/integreatly-operator.v2.0.0.clusterserviceversion.yaml#L44
  * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/deploy/olm-catalog/integreatly-operator/integreatly-operator-2.0.0/integreatly-operator.v2.0.0.clusterserviceversion.yaml#L273
* Push csv to your registry
  * `make push/csv  REPO=<your_registry> QUAY_USERNAME=<your_registry> QUAY_PASSWORD=<password>`
* Install operator source with your registry
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: RHMI
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: RHMI Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: ${REPO}
  type: appregistry
```
* Create operator namespace and setup secrets
  * `make cluster/prepare/project && make cluster/prepare/smtp && make cluster/prepare/dms && make cluster/prepare/pagerduty`
* Install RHMI from operator hub to the `redhat-rhmi-operator` namespace
### Upgradel via OLM
* Checkout master 
* Change makefile `ORG` to your own quay.io registry
  * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/Makefile#L3
* Push csv to your registry
  * `make push/csv  REPO=<your_registry> QUAY_USERNAME=<your_registry> QUAY_PASSWORD=<password>`
* Install operator source with your registry
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: RHMI
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: RHMI Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: ${REPO}
  type: appregistry
```
* Create operator namespace and setup secrets
  * `make cluster/prepare/project && make cluster/prepare/smtp && make cluster/prepare/dms && make cluster/prepare/pagerduty`
* Install RHMI from operator hub to the `redhat-rhmi-operator` namespace
* When installation has complete, checkout this branch
* Generate new csv 
  * Change makefile `PREVIOUS_TAG` to `2.0.0`
  * Change makefile `TAG` to `2.1.0`
  * `make gen/csv`
* Build and push RHMI image to your own registry
  * `make image/build/push`
* Replace image in new csv with built image
  * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/deploy/olm-catalog/integreatly-operator/integreatly-operator-2.0.0/integreatly-operator.v2.0.0.clusterserviceversion.yaml#L44
   * https://github.com/integr8ly/integreatly-operator/blob/8eee7ace6c571569d62eddca93cc8d18b2daced6/deploy/olm-catalog/integreatly-operator/integreatly-operator-2.0.0/integreatly-operator.v2.0.0.clusterserviceversion.yaml#L273
  * Check that the `replaces` is correct
* Push csv to your registry
  * `make push/csv  REPO=<your_registry> QUAY_USERNAME=<your_registry>  QUAY_PASSWORD=<password>`
* Remove `status` field of operator source created above.
* Observe that there is a available upgrade for RHMI
  * Would need to manually approve install plan to trigger upgrade
* Observe an upgrade for 3scale is triggered and succeeds
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Verified independently on a cluster by reviewer

### CODE REVIEW

**Comment by KevFan** [comment]
@jjaferson @philbrookes Noticed something visually wrong after an upgrade for 3scale, where the version is not updated in the RHMI CR afterwards. I think it's due to this config map, where we get and set the product version from is not updated by the 3scale operator to the latest version. 
https://github.com/integr8ly/integreatly-operator/blob/2a3f3198b4dc22c06fdc8cf57ff85e5006314498/pkg/products/threescale/reconciler.go#L894-L900
Wondering what's your opinion on this? Update the config map to always match the version of 3scale or remove this part of the code and just always use the hardcoded version? :thinking: 

Screenshot from 2020-04-06 17-03-00

**Comment by davidffrench** [comment]
@KevFan Looks like it needs a rebase

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/integr8ly/integreatly-operator
 go 1.13
 
 require (
-	github.com/3scale/3scale-operator v0.4.0
+	github.com/3scale/3scale-operator v0.2.1-0.20200327122530-d376df6cc71f
```
Comment (davidffrench) [inline go.mod:6]: Curious why not use the `v0.5.0` tag here? https://github.com/3scale/3scale-operator/releases/tag/v0.5.0

Comment (KevFan) [reply go.mod:6]: Oh didn't realise they've tagged a 0.5.0 release now. They didn't have that tag when I was testing the update. I'll update the PR to use that tag now :+1:

**Comment by davidffrench** [comment]
/lgtm
/approve

Thanks @KevFan . I have not verified, approving based on code review and previous verification

**Pull request merged**